lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -150,9 +150,13 @@ public function bindParam($column, &$variable, $type = null, $length = null)
      */
     public function closeCursor()
     {
-        if ($this->stmt) {
```

Comment:
Also reusing prepared statements does currently not work as the binding by reference does not work. We need to modify `bindParam()` to this:

```php
public function bindParam($column, &$variable, $type = null, $length = null)
{
    if (!is_numeric($column)) {
        throw new SQLSrvException("sqlsrv does not support named parameters to queries, use question mark (?) placeholders instead.");
    }

    $this->values[$column] = $variable;

    if ($type === \PDO::PARAM_LOB) {
        $this->params[$column-1] = array(&$this->values[$column], SQLSRV_PARAM_IN, SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_BINARY), SQLSRV_SQLTYPE_VARBINARY('max'));
    } else {
        $this->params[$column-1] =& $this->values[$column];
    }
}
```
Also introduces a new private property `$values` that keeps track of parameter references. Otherwise using different parameters won't work and the driver will always use those from the first binding.